src/OpenTelemetry/Metrics/Tags.cs

```diff
@@ -20,10 +20,23 @@ namespace OpenTelemetry.Metrics
 {
     internal readonly struct Tags : IEquatable<Tags>
     {
+        private readonly int hashCode;
+
         public Tags(string[] keys, object[] values)
         {
             this.Keys = keys;
             this.Values = values;
+
+            unchecked
+            {
+                var hash = 17;
+                for (int i = 0; i < this.Keys.Length; i++)
+                {
+                    hash = (hash * 31) + this.Keys[i].GetHashCode() + this.Values[i]?.GetHashCode() ?? 0;
+                }
+
+                this.hashCode = hash;
+            }
```

Comment:
@tgrieger-sf No because the the keys and values never do change. This struct is internal and instances of it are only used within our AggregatorStore for looking up metric points https://github.com/open-telemetry/opentelemetry-dotnet/blob/a19d106e0c40b60f5d0032999c315a8f0b8c5159/src/OpenTelemetry/Metrics/AggregatorStore.cs#L33.

Reply:
If that's the case, is there any benefit in using `ReadOnlySpan<string/object>` for Keys and Values to make it explicit?

Reply:
`ReadOnlySpan` is a `ref struct` and cannot be used a valid type for a dictionary as it has to live on stack.